src/apps/answers/service.py

```diff
@@ -744,8 +765,15 @@ async def get_flow_submission(
 
         answer_result: list[ActivityAnswer] = []
 
+        source_subject_id_answer_index_map: dict[uuid.UUID, list[int]] = {}
+
```

Comment:
Here you can construct this map as `defaultdict(list)` and just call bellow `source_subject_id_answer_index_map[answer.source_subject_id].append(i)` with check if already exists

source_subject_id_answer_index_map: dict[uuid.UUID, list[int]] = defaultdict(list)

if answer.source_subject_id:
    source_subject_id_answer_index_map[answer.source_subject_id].append(i)

Reply:
Sweet tip. Thanks @rcmerlo!